config.go

```diff
@@ -0,0 +1,135 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/packer/common"
+	"github.com/hashicorp/packer/helper/communicator"
+	"github.com/hashicorp/packer/helper/config"
+	"github.com/hashicorp/packer/packer"
+	"github.com/hashicorp/packer/template/interpolate"
+	"strconv"
+)
+
+const Unspecified = -1
+
+type VMCustomParams struct {
+	Cpu_sockets int
+	Ram         int
+	// TODO: add more options
+}
+
+type VMParams struct {
+	Url            string
+	Username       string
+	Password       string
+	Dc_name        string
+	Folder_name    string
+	Vm_source_name string
+	Vm_target_name string
+}
+
+type Config struct {
+	common.PackerConfig `mapstructure:",squash"`
+	communicator.Config `mapstructure:",squash"`
+
+	Url            string `mapstructure:"url"`
+	Username       string `mapstructure:"username"`
+	Password       string `mapstructure:"password"`
+	Ssh_username   string `mapstructure:"ssh_username"`
+	Ssh_password   string `mapstructure:"ssh_password"`
+
+	Dc_name        string `mapstructure:"dc_name"`
+	Vm_source_name string `mapstructure:"template"`
+	Vm_target_name string `mapstructure:"vm_name"`
+
+	Cpu_sockets    string `mapstructure:"cpus"`
+	Shutdown_cmd   string `mapstructure:"shutdown_command"`
+	Ram            string `mapstructure:"RAM"`
+
+	// internal
+	vm_params VMParams
+	vm_custom VMCustomParams
+
+	ctx      interpolate.Context
+}
+
+
+func NewConfig(raws ...interface{}) (*Config, []string, error) {
+	c := new(Config)
+	err := config.Decode(c, &config.DecodeOpts{
+		Interpolate:        true,
+		InterpolateContext: &c.ctx,
+		InterpolateFilter: &interpolate.RenderFilter{
+			Exclude: []string{
+				"shutdown_command",
```

Comment:
What's the reason to exclude this parameter?
`shutdown` command requires `sudo`, so root password is required. I usually specify it like
```
"shutdown_command": "echo '{{user `password`}}' | sudo -S shutdown -P now"
```